Applications/Cxx/gdcmanon.cxx

```diff
@@ -173,28 +156,79 @@ static bool AnonymizeOneFile(gdcm::Anonymizer &anon, const char *filename, const
   gdcm::FileMetaInformation &fmi = file.GetHeader();
   fmi.Clear();
 
-  gdcm::Writer writer;
-  if (strcmp(outfilename, "stdout") == 0) {
-    writer.SetStream(std::cout);
-  } else {
-    writer.SetFileName(outfilename);
-  }
   writer.SetFile( file );
-  if( !writer.Write() )
-    {
-    std::cerr << "Could not Write : " << outfilename << std::endl;
-    if( strcmp(filename,outfilename) != 0 )
-      {
-      gdcm::System::RemoveFile( outfilename );
+
+  return true;
+}
+
+static bool AnonymizeOneFile(gdcm::Anonymizer &anon, const char *filename, const char *outfilename, bool continuemode) {
+    gdcm::Reader reader;
+    reader.SetFileName(filename);
+
+    if (!reader.Read()) {
+      std::cerr << "Could not read : " << filename << std::endl;
+      if (continuemode) {
+        std::cerr << "Skipping from anonymization process (continue mode)." << std::endl;
+        return true;
+      } else {
+        std::cerr << "Check [--continue] option for skipping files." << std::endl;
+        return false;
       }
-    else
-      {
-      std::cerr << "gdcmanon just corrupted: " << filename << " for you (data lost)." << std::endl;
+    }
+
+    gdcm::Writer writer;
+    writer.SetFileName(outfilename);
+    if (!AnonymizeOneFile(anon, reader, writer, filename)) {
+      return false;
+    }
+
+    if (!writer.Write()) {
+      std::cerr << "Could not Write : " << outfilename << std::endl;
+      if (strcmp(filename, outfilename) != 0) {
+        gdcm::System::RemoveFile(outfilename);
+      } else {
+        std::cerr << "gdcmanon just corrupted: " << filename << " for you (data lost)." << std::endl;
       }
+      return false;
+    }
 
-    return false;
+    return true;
+}
+
+static void anonymize_network_files(gdcm::Anonymizer &anon, unsigned short port)
+{
+  io_service ioservice;
+  tcp::endpoint endpoint(tcp::v4(), port);
+  tcp::acceptor acceptor(ioservice, endpoint);
+  std::cout << "Listening for connections on: " << port << std::endl;
+
+  while (true) {
+    tcp::iostream socket_stream;
+    boost::system::error_code connection_error;
+    acceptor.accept(*socket_stream.rdbuf(), connection_error);
+    fixed_istream_buffer seek_buffer(socket_stream, 512);
+    std::istream buffered_stream(&seek_buffer);
+
+    if (!connection_error) {
+      gdcm::Reader reader;
+      reader.SetStream(buffered_stream);
+      if(!reader.Read()) {
+        std::cerr << "Could not read tcp stream" << std::endl;
+        continue;
```

Comment:
This should probably be a process ending error.

Reply:
Since we only process one file per connection this just means stop this connection and process the next file connection (and so is equivalent to --continue in file mode).